Chapter02/recipe-0012/README.md

```diff
@@ -0,0 +1,107 @@
+# Dealing with platform-dependent source code
+
+Ideally we should avoid platform-dependent source code but sometimes we have no choice - in particular
+if we are given a code to configure and compile which we have not written.
+In this recipe we will demonstrate how to use CMake to conditionally compile source code section
+depending on the operating system.
+
+
+## Ingredients
+
+- A C++ compiler
+- CMake (any version higher than 3.0)
+
+
+## How to do it
+
+For this example we will use a slight modification of the previous recipe.
+
+Let us consider the following `hello-world.cpp` source code:
+
+```cpp
+#include <cstdlib>
+#include <iostream>
+#include <string>
+
+std::string say_hello() {
+#ifdef IS_WINDOWS
+  return std::string("Hello from Windows!");
+#elif IS_LINUX
+  return std::string("Hello from Linux!");
+#elif IS_MAC
+  return std::string("Hello from Mac!");
+#else
+  return std::string("Hello from an unknown system!");
+#endif
+}
+
+int main() {
+  std::cout << say_hello() << std::endl;
+  return EXIT_SUCCESS;
+}
+```
+
+We will configure this code using the following `CMakeLists.txt`:
+
+```cmake
+# set minimum cmake version
+cmake_minimum_required(VERSION 3.0 FATAL_ERROR)
+
+# project name and language
+project(recipe-0011 CXX)
+
+# let the preprocessor know about the system name
+if(${CMAKE_SYSTEM_NAME} STREQUAL "Linux")
```

Comment:
1. Same as before, no need for `${}` syntax
2. I would rework this example to use [`target_compile_definitions`](https://cmake.org/cmake/help/v3.0/command/target_compile_definitions.html) instead of `add_definitions`. Using global commands such as `add_definitions`, `include_directories` and `link_directories` is basically deprecated in CMake >= 3.0

Reply:
Thanks - I shall adapt and understand why the latter is probably better but also want to note that if they are deprecated, the CMake documentation does not seem to advertise this fact.

Reply:
No, CMake docs are lacking on many things, sadly

Reply:
This is where we come in! :-) Thanks again for making me aware of these developments.